Investigate and fix zarr/Blosc issues

[Python 3.7 CI](https://github.com/NCAR/xpersist/runs/1811052841?check_suite_focus=true) has been failing for the last few days due to some intermittent Blosc issues:

```bash
tests/test_core.py::test_save_as_zarr FAILED                             [100%]

=================================== FAILURES ===================================
______________________________ test_save_as_zarr _______________________________

    def test_save_as_zarr():
        with TemporaryDirectory() as local_store:
>           dsp = xp.persist_ds(func, name='test-dset', path=local_store, format='zarr')(10)

tests/test_core.py:108: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
xpersist/core.py:143: in __call__
    ds.to_zarr(self._cache_file, consolidated=True)
/usr/share/miniconda/envs/xpersist-dev/site-packages/xarray/core/dataset.py:1756: in to_zarr
    region=region,
/usr/share/miniconda/envs/xpersist-dev/site-packages/xarray/backends/api.py:1477: in to_zarr
    write_region=region,
/usr/share/miniconda/envs/xpersist-dev/site-packages/xarray/backends/zarr.py:286: in open_group
    import zarr
/usr/share/miniconda/envs/xpersist-dev/site-packages/zarr/__init__.py:2: in <module>
    from zarr.codecs import *
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    from numcodecs import *
>   from numcodecs import get_codec, Blosc, Zlib, Delta, AsType, BZ2
E   ImportError: cannot import name 'Blosc'

/usr/share/miniconda/envs/xpersist-dev/site-packages/zarr/codecs.py:3: ImportError
----------------------------- Captured stdout call -----------------------------
writing cache file: /tmp/tmpkvn1gvsz/test-dset.zarr
=============================== warnings summary ===============================
tests/test_core.py::test_xpersist_actions
  /usr/share/miniconda/envs/xpersist-dev/lib-python/3/importlib/_bootstrap.py:228: UserWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 96 from PyObject
    return f(*args, **kwds)
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -36,8 +36,8 @@ jobs:
           auto-update-conda: false
           python-version: ${{ matrix.python-version }}
           environment-file: ci/environment.yml
-          mamba-version: '*'
-          use-mamba: true
+          # mamba-version: '*'
+          # use-mamba: true
           miniforge-variant: Mambaforge
           miniforge-version: latest
 
@@ -84,8 +84,8 @@ jobs:
           auto-update-conda: false
           python-version: 3.8
           environment-file: ci/environment-upstream-dev.yml
-          mamba-version: '*'
-          use-mamba: true
+          # mamba-version: '*'
+          # use-mamba: true
           miniforge-variant: Mambaforge
           miniforge-version: latest
 

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -37,9 +37,6 @@ jobs:
           python-version: ${{ matrix.python-version }}
           environment-file: ci/environment.yml
           mamba-version: '*'
-          use-mamba: true
-          miniforge-variant: Mambaforge
-          miniforge-version: latest
 
       - name: Install intake-esm
         run: |
@@ -85,9 +82,6 @@ jobs:
           python-version: 3.8
           environment-file: ci/environment-upstream-dev.yml
           mamba-version: '*'
-          use-mamba: true
-          miniforge-variant: Mambaforge
-          miniforge-version: latest
 
       - name: Set environment variables
         run: |

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -36,10 +36,9 @@ jobs:
           auto-update-conda: false
           python-version: ${{ matrix.python-version }}
           environment-file: ci/environment.yml
-          # mamba-version: '*'
-          # use-mamba: true
+          mamba-version: '*'
+          use-mamba: true
           miniforge-variant: Mambaforge
-          miniforge-version: latest
 
       - name: Install intake-esm
         run: |
@@ -84,10 +83,9 @@ jobs:
           auto-update-conda: false
           python-version: 3.8
           environment-file: ci/environment-upstream-dev.yml
-          # mamba-version: '*'
-          # use-mamba: true
+          mamba-version: '*'
+          use-mamba: true
           miniforge-variant: Mambaforge
-          miniforge-version: latest
 
       - name: Set environment variables
         run: |

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -28,7 +28,7 @@ jobs:
         with:
           fetch-depth: 0 # Fetch all history for all branches and tags.
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - uses: conda-incubator/setup-miniconda@master
         with:
           channels: conda-forge
           channel-priority: strict
@@ -37,6 +37,9 @@ jobs:
           python-version: ${{ matrix.python-version }}
           environment-file: ci/environment.yml
           mamba-version: '*'
+          use-mamba: true
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
 
       - name: Install intake-esm
         run: |
@@ -73,7 +76,7 @@ jobs:
         with:
           fetch-depth: 0 # Fetch all history for all branches and tags.
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - uses: conda-incubator/setup-miniconda@master
         with:
           channels: conda-forge
           channel-priority: strict
@@ -82,6 +85,9 @@ jobs:
           python-version: 3.8
           environment-file: ci/environment-upstream-dev.yml
           mamba-version: '*'
+          use-mamba: true
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
 
       - name: Set environment variables
         run: |

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -31,7 +31,7 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v2
         with:
           channels: conda-forge
-          mamba-version: '*'
+          # mamba-version: '*'
           channel-priority: strict
           activate-environment: intake-esm-dev
           auto-update-conda: false
@@ -76,7 +76,7 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v2
         with:
           channels: conda-forge
-          mamba-version: '*'
+          # mamba-version: '*'
           channel-priority: strict
           activate-environment: intake-esm-dev
           auto-update-conda: false

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -1,11 +1,7 @@
 name: CI
 on:
   push:
-    branches:
-      - '*'
   pull_request:
-    branches:
-      - '*'
   schedule:
     - cron: '0 0 * * *' # Daily “At 00:00”
   workflow_dispatch: # allows you to trigger manually
@@ -15,6 +11,7 @@ jobs:
     name: python-${{ matrix.python-version }}
     if: github.repository == 'intake/intake-esm'
     runs-on: ubuntu-latest
+    timeout-minutes: 30
     defaults:
       run:
         shell: bash -l {0}
@@ -39,11 +36,11 @@ jobs:
           activate-environment: intake-esm-dev
           auto-update-conda: false
           python-version: ${{ matrix.python-version }}
+          environment-file: ci/environment.yml
 
-      - name: Set up conda environment
+      - name: Install intake-esm
         run: |
-          mamba env update -f ci/environment.yml
-          python -m pip install -e .
+          python -m pip install -e . --no-deps --force-reinstall
           conda list
 
       - name: Run Tests
@@ -63,6 +60,7 @@ jobs:
     name: upstream-dev
     if: github.repository == 'intake/intake-esm'
     runs-on: ubuntu-latest
+    timeout-minutes: 30
     defaults:
       run:
         shell: bash -l {0}
@@ -83,15 +81,15 @@ jobs:
           activate-environment: intake-esm-dev
           auto-update-conda: false
           python-version: 3.8
+          environment-file: ci/environment-upstream-dev.yml
 
       - name: Set environment variables
         run: |
           echo "XARRAY_BACKEND_API=v2" >> $GITHUB_ENV
 
-      - name: Set up conda environment
+      - name: Install intake-esm
         run: |
-          mamba env update -f ci/environment-upstream-dev.yml
-          python -m pip install -e .
+          python -m pip install -e . --no-deps --force-reinstall
           conda list
 
       - name: Run Tests

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -31,12 +31,15 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v2
         with:
           channels: conda-forge
-          # mamba-version: '*'
           channel-priority: strict
           activate-environment: intake-esm-dev
           auto-update-conda: false
           python-version: ${{ matrix.python-version }}
           environment-file: ci/environment.yml
+          mamba-version: '*'
+          use-mamba: true
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
 
       - name: Install intake-esm
         run: |
@@ -76,12 +79,15 @@ jobs:
       - uses: conda-incubator/setup-miniconda@v2
         with:
           channels: conda-forge
-          # mamba-version: '*'
           channel-priority: strict
           activate-environment: intake-esm-dev
           auto-update-conda: false
           python-version: 3.8
           environment-file: ci/environment-upstream-dev.yml
+          mamba-version: '*'
+          use-mamba: true
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
 
       - name: Set environment variables
         run: |

--- a/ci/environment-upstream-dev.yml
+++ b/ci/environment-upstream-dev.yml
@@ -2,6 +2,7 @@ name: intake-esm-dev
 channels:
   - conda-forge
 dependencies:
+  - python=*=*cp*
   - boto3
   - cartopy
   - cftime

--- a/ci/environment-upstream-dev.yml
+++ b/ci/environment-upstream-dev.yml
@@ -3,21 +3,27 @@ channels:
   - conda-forge
 dependencies:
   - boto3
+  - cartopy
   - cftime
   - codecov
   - fastprogress>= 0.2.6
   - h5netcdf>=0.8.1
   - ipython
+  - matplotlib
   - netcdf4>=1.5.5
+  - pandas
   - pip
   - pre-commit
   - psutil
+  - pyproj
   - pytest
   - pytest-cov
   - pytest-icdiff
   - pytest-xdist
+  - regionmask
   - scipy
   - toolz
+  - xgcm
   - zarr>=2.5
   - pip:
       - git+https://github.com/intake/filesystem_spec.git

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -2,6 +2,7 @@ name: intake-esm-dev
 channels:
   - conda-forge
 dependencies:
+  - python=*=*cp*
   - cftime
   - codecov
   - fastprogress>= 0.2.6

--- a/tests/test_merge_util.py
+++ b/tests/test_merge_util.py
@@ -89,7 +89,11 @@ def test_open_asset_error(path, data_format, error):
             {'anon': True},
         ),
         ('s3://ncar-cesm-lens/lnd/monthly/cesmLE-20C-FSNO.zarr', 'zarr', {'anon': True}),
-        ('gs://cmip6/AerChemMIP/BCC/BCC-ESM1/ssp370/r1i1p1f1/Amon/pr/gn/', 'zarr', {'anon': True}),
+        (
+            'gs://cmip6/CMIP6/AerChemMIP/BCC/BCC-ESM1/ssp370/r1i1p1f1/Amon/pr/gn/v20190624/',
+            'zarr',
+            {'anon': True},
+        ),
         (
             f'file://{here}/sample_data/cmip/CMIP6/CMIP/BCC/BCC-ESM1/piControl/r1i1p1f1/Amon/tasmax/gn/v20181214/tasmax/tasmax_Amon_BCC-ESM1_piControl_r1i1p1f1_gn_185001-230012.nc',
             'netcdf',

